tiledb/ml/readers/pytorch_sparse.py

```diff
@@ -111,13 +112,14 @@ def __iter__(self) -> Iterator[DataType]:
         # Loop over batches
         for offset in range(iter_start, iter_end, self.batch_size):
             # Yield the next training batch
-            x_batch = self.x[offset : offset + self.batch_size]
-            y_batch = self.y[offset : offset + self.batch_size]
+            parallel_batches = run_io_tasks_in_parallel(
+                (self.x, self.y), self.batch_size, offset
+            )
 
             x_coords = []
             for i in range(0, self.x.schema.domain.ndim):
                 dim_name = self.x.schema.domain.dim(i).name
-                x_coords.append(x_batch[dim_name])
+                x_coords.append(parallel_batches[0]._result[dim_name])
```

Comment:
1. As its name implies, `Future._result` is a "private" attribute. The right way is to call `Future.result()` which waits until the Future finishes and reraises any exception.
2. `run_io_tasks_in_parallel` can return results instead of futures. This way, the only change in this module can be one line:
```
x_batch, y_batch = run_io_tasks_in_parallel((self.x, self.y), self.batch_size, offset)
# everything below keeps using x_batch, y_batch.
```